salt/returners/elasticsearch_return.py

```diff
@@ -405,6 +405,8 @@ def get_load(jid):
     data = __salt__['elasticsearch.document_get'](index=index,
                                                   id=jid,
                                                   doc_type=doc_type)
+    if isinstance(data, dict):
+        return data
```

Comment:
I would actually simply rewrite this function as:

```
try:
    return salt.utils.json.loads(data)
except SomeError:
    return data or {}
```

Reply:
Following back up, I have done more digging in the code and the documentation says that SaltStack supports Elasticsearch 1.5+ and the code for supporting the `master_job_cache` in Elastic landed in Salt 2015.8.1. The `salt.modules.elasticsearch` at that time (1) combined with the version of the python library (2) for even that old of an Elastic instance would still have returned at `dict()`. I think the code above makes the correct code path seem exceptional and we should remove handling of strings all together since this appears to have never worked.

Any error handling should be done at the salt module by ensuring that `__salt__['elasticsearch.document_get']()` should always return a `dict()`.  Plus this is easier to ensure with a test case.

So I think the `get_load()` method should just `return data or {}`. 

#### References
1. https://github.com/saltstack/salt/blob/2015.8/salt/modules/elasticsearch.py#L249
2. https://github.com/elastic/elasticsearch-py/blob/1.5.0/elasticsearch/transport.py#L331

Reply:
> Hey, thanks for the PR! @calvinhp this should actually be fairly easy (and worthwhile) to write a unit test for. You can simply follow this pattern found here
> https://github.com/saltstack/salt/blob/2018.3/tests/unit/states/test_elasticsearch.py#L90

Since the `get_load()` method is using the `salt.modules.elasticsearch.document_get()` method. I think we should be ensuring that this method always returns a `dict()`. This has tests:

https://github.com/saltstack/salt/blob/develop/tests/unit/modules/test_elasticsearch.py#L676

Do you have a suggestion for what I should add that would ensure the job cache methods work?